### DISCUSSION TITLE
A mask consisting of several numbers in a row with a colon

### DISCUSSION THREAD

**Discussion by whodatilya** [uNmAnNeR/imaskjs#1059] [Q&A]
Hello everyone! I've been struggling with one problem for quite some time now:
There is a field, it consists of 2 blocks:
First: 
6 or 7 digits, 
Second:
1 or 3 digits.
It looks like:
_first:second._

I want to implement the following logic:
After the user has entered 6 digits of the first block, by default a colon should be automatically inserted and the cursor should move beyond the colon, however, if the user needs to enter 7 digits in the first block, then after the cursor has jumped over the colon, the user can press the key to return to the left and add another number.

At the moment I have the mask 
**"000000[0]:0[00]",**
but it doesn't work because the 7th digit in the first block is added immediately. Help me please:(

**Comment by alessiobianchini** [comment]
Hi @whodatilya, what framework are you using?
